sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -852,23 +835,171 @@ case class GpuRegExpReplace(
 
   override def doColumnar(
       strExpr: GpuColumnVector,
-      searchExpr: GpuScalar,
-      replaceExpr: GpuScalar): ColumnVector = {
-    strExpr.getBase.replaceRegex(cudfRegexPattern, replaceExpr.getBase)
-  }
+      searchExpr: GpuColumnVector,
+      replaceExpr: GpuScalar): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
 
   override def doColumnar(numRows: Int, val0: GpuScalar, val1: GpuScalar,
       val2: GpuScalar): ColumnVector = {
-    withResource(GpuColumnVector.from(val0, numRows, srcExpr.dataType)) { val0Col =>
+    withResource(GpuColumnVector.from(val0, numRows, first.dataType)) { val0Col =>
       doColumnar(val0Col, val1, val2)
     }
   }
 
+}
+
+case class GpuRegExpReplace(
+    srcExpr: Expression,
+    searchExpr: Expression,
+    replaceExpr: Expression,
+    cudfRegexPattern: String)
+  extends GpuRegExpTernaryBase with ImplicitCastInputTypes {
+
+  override def dataType: DataType = srcExpr.dataType
```

Comment:
Was surprised that this was not implemented in `GpuRegExpTernaryBase`.  When would we expect `dataType` for any regexp-like operator to _not_ be `StringType`?  Definitely no change necessary here and totally fine if this is just parroting the Spark source for consistency, but I'm curious if this really needs to be overridden here rather than specified in the new base class as always being `StringType`.

Reply:
Fixed.